api/model/mixins/Dictable.py

```diff
@@ -20,21 +22,27 @@ def as_dict(self, include_pk=True):
         fields = {c.name: getattr(self, c.name) for c in self.__table__.columns
                   if c.name not in excluded_keys}
 
+        exclude_dict_fields_list = override_exclude_dict_fields or self.exclude_dict_fields
+
         for relationship in self.__mapper__.relationships:
             rel = relationship.key
-            if rel not in self.exclude_dict_fields:
+            if rel not in exclude_dict_fields_list and relationship.entity.class_ not in models_hit:
                 rel_map = getattr(self, rel)
                 if isinstance(rel_map, InstrumentedList):
-                    fields[rel] = [self._dict_or_collapsed(el) for el in rel_map]
+                    fields[rel] = [self._dict_or_collapsed(el, models_hit) for el in rel_map]
                 else:
                     rel_fixed_name = rel.replace('lookup_', '') if rel.startswith('lookup_') else rel
-                    fields[rel_fixed_name] = self._dict_or_collapsed(rel_map)
+                    fields[rel_fixed_name] = self._dict_or_collapsed(rel_map, models_hit)
 
         return fields
 
+    def as_dict(self, include_pk=True, override_exclude_dict_fields=[]):
+        return self._as_dict_recur([self.__class__], include_pk, override_exclude_dict_fields)
+
     @staticmethod
-    def _dict_or_collapsed(obj):
+    def _dict_or_collapsed(obj, models_hit=[]):
```

Comment:
Mutable default arguments are bad. 
This works better. You could also use an if conditional
`
def _dict_or_collapsed(obj, models_hit=None):
     models_hit = models_hit or []
`